core/system_chaincode/noop/chaincode.go

```diff
@@ -0,0 +1,105 @@
+/*
+ Copyright Digital Asset Holdings, LLC 2016 All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package noop
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric/core/chaincode/shim"
+	ld "github.com/hyperledger/fabric/core/ledger"
+	"github.com/hyperledger/fabric/protos"
+)
+
+var logger = shim.NewLogger("noop")
+
+type ledgerHandler interface {
+	GetTransactionByUUID(txUUID string) (*protos.Transaction, error)
+}
+
+// SystemChaincode is type representing the chaincode
+// It has a ledgerHandler to be able to get TXs by ID
+type SystemChaincode struct {
+	ledgerH ledgerHandler
+}
+
+// Init initailizes the system chaincode
+func (t *SystemChaincode) Init(stub *shim.ChaincodeStub, function string, args []string) ([]byte, error) {
+	var ledger, err = ld.GetLedger()
+	if nil != err {
+		return nil, err
+	}
+	t.ledgerH = ledger
```

Comment:
To be honest, this is just for testing. I wanted to mock the ledger somehow. If I cannot use Init, I can also check if t.ledgerHandler is nil or not. If it is nil, I can get the ledger singleton.